### ISSUE TITLE
generate StructTypes traits

### ISSUE THREAD

**Issue by Roger-luo** [Roger-luo/Configurations.jl#43]
we should be able to generate overloads for StructTypes automatically, but need to think about whether do it here or in KungIChi since I'd like to keep this package as minimal as possible.

**Comment by singularitti** [comment]
In the docs there said:

> And we have plans of supporting StructTypes traits by default once [JuliaData/StructTypes#53](https://github.com/JuliaData/StructTypes.jl/issues/53) is figured out.

Since JuliaData/StructTypes#53 is closed, has [Configurations.jl](https://github.com/Roger-luo/Configurations.jl) support it yet?

**Comment by Roger-luo** [comment]
not really, as you see it's not closed. I currently don't have much time to work on this package, but please feel free to work on a PR if you need this feature.
